packages/wallets/src/components/UIProvider/UIProvider.tsx

```diff
@@ -0,0 +1,51 @@
+import React, { createContext, useContext, useState } from 'react';
+
+type Region = 'EU' | 'Non-EU';
+
+type TUIState = {
+    region: Region;
+};
+
+type TUIContext = {
+    setUIState: (key: keyof TUIState, value: TUIState[keyof TUIState]) => void;
```

Comment:
```suggestion
    setUIState: (key: keyof TUIState, value: TUIState[typeof key]) => void;
```
@thisyahlen-deriv  You have to use it like this to prevent using cases from another property type. For example, if we have 2 properties, region and testRegion, region can be 'EU' | 'Non-EU', and testRegion can be 'testEU' | 'testNon-EU'. With your core when the developer use key region he can add 4 cases for value  'EU' | 'Non-EU' | 'testEU' | 'testNon-EU', but it's not the right, you have to avoid use cases from other types